### ISSUE TITLE
Macro not expanded when part of a dnat destination

### ISSUE THREAD

**Issue by lx1** [FoobarOy/foomuuri#50]
If I have something like:

```
macro {
  server        192.168.20.246
}
dnat {
  iifname enp1s0 tcp 1234 dnat to server
}
```

then foomuuri check gives success. But if I add the port in the dnat destination, for ex.

```
macro {
  server        192.168.20.246
}
dnat {
  iifname enp1s0 tcp 1234 dnat to server:123
}
```

the foomuuri check complains:
Error: File /etc/foomuuri/rules.conf line nn: Invalid IP address in "to": server:123

This is only an example of a more general problem where macros seem to be expanded only when they appear as a whole word, delimited by whitespace. But there are cases, like this one, where it would be useful for a macro to appear in some more complex expression.

**Comment by kimheino** [comment]
There are at least following cases:

```
server:123 
[server]:123  
server/8
server/-64
-server
-server/8
-server/-64
serverA-serverB
portA-portB
...
server/mask # both server and mask are macros
...
```

To make things harder, server could be defined as multiple IPs.

Are there other cases? 

I'll have to think about this, but I think I'll only support limited set of these. This is function expand_single_line()

**Comment by lx1** [comment]
I realized that it was expand_single_line() but I'm not expert in python. I saw that the line is previously split in an array of strings separated by whitespaces when it comes to expand macros. The macro is searched within these strings with an entire match. Maybe the match could be made on word boundaries and not on the whole string. But honestly these are the first times I try foomuuri and I don't know if this approach can lead to some unwanted match. Based on the cases you mention, I would tell that if the match is done as a whole word with non-word boundaries, it should work good, but I feel don't have enough experience.
